graylog2-server/src/main/java/org/graylog2/configuration/converters/JavaDurationConverter.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.configuration.converters;
+
+import com.github.joschi.jadconfig.Converter;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.TimeUnit;
+
+public class JavaDurationConverter implements Converter<Duration> {
+    @Override
+    public Duration convertFrom(String value) {
+        final com.github.joschi.jadconfig.util.Duration jadDuration = com.github.joschi.jadconfig.util.Duration.parse(value);
+
+        final ChronoUnit chronoUnit = toChronoUnit(jadDuration.getUnit());
+        return Duration.of(jadDuration.getQuantity(), chronoUnit);
+    }
+
+    @Override
+    public String convertTo(Duration value) {
+        return value.toString();
```

Comment:
My suggestion is probably wrong, if `Duration` gives us the unit, we should use that.

Reply:
Since this is only used in tests, it might also ne be important. :smile:

Reply:
A millisecond long might overflow for very long durations.
And it's not only used in tests. the leader-election timeouts are configured this way.
I thought it's time we move towards a modern java.time implementation..

Reply:
@mpfz0r The `#convertTo()` method is only used in tests. Sorry, I was writing notes during a call with @thll, that's why it's a bit unstructured.

The main issue is that this converter converts **from** a jadconfig duration string but converts **to** a java time duration string. We need to check what we can do to make this consistent.

Reply:
Ah, yeah that's right. The problem is that iso8601 durations are more complicated than the jadconfig ones.
e.g. `PT3H5S` cannot be expressed in jadconfig.
It would need to be converted to the smallest unit.
so that would be  `10805 seconds`

Reply:
Addressed in https://github.com/Graylog2/graylog2-server/pull/11709